### PR TITLE
Ensure numberClick with same slots doesn't do anything

### PR DESCRIPTION
**Pull request by kaduvert**

### CODE REVIEW

File: lib/Window.js [hunk 1]
```diff
@@ -149,6 +149,7 @@ module.exports = (Item, registry) => {
       const { item } = click
       const hotbarSlot = this.hotbarStart + click.mouseButton
       const itemAtHotbarSlot = this.slots[hotbarSlot]
+      if (item === itemAtHotbarSlot) return
```
Comment (rom1504) [inline lib/Window.js]: is comparing by reference appropriate here?

Comment (kaduvert) [reply lib/Window.js]: as long as references are passed, which currently is the case.
`Item.equal` doesn't compare slot index, but that's important here

i also added a test case for this, so it's ok to compare by reference because the test passes

Comment (kaduvert) [reply lib/Window.js]: so i changed it anyways so it doesn't break in the future :)

**Comment by rom1504** [comment]
ok, can you explain why clicking on the same slot would do nothing?

**Comment by kaduvert** [comment]
numberClicks (mode 2) is for quickly swapping single itemStacks between slots and swapping an item with one in the same slot should do nothing.

this pr is mainly to cover this edge case.
no one would really do such a click, but it can occur in testing where slots and mouseButtons are random.
to not break tests, i implemented this to be sure
